### PR TITLE
Added proxy tests

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "btoa": "1.1.2",
     "cheerio": "0.22.0",
     "crypto-js": "3.1.6",
+    "http-proxy": "1.15.1",
```
Comment (czardoz) [inline package.json]: This should be in dev deps

File: test/integration/sanity/proxy-http.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+describe('sanity test', function () {
```
Comment (czardoz) [inline test/integration/sanity/proxy-http.test.js]: Great. Now we need to go to the next step.

How do we verify whether our proxy was actually used?

File: test/integration/sanity/proxy-http.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    var proxy = require('http-proxy'),
+
+        server = new proxy.createProxyServer({target: 'https://echo.getpostman.com/get'}),
+
+        testrun;
+
+    before(function (done) {
+        var fakeProxyManager = {
+            getProxyConfiguration: function (url, callback) {
+                callback(null, server.options);
```
Comment (czardoz) [inline test/integration/sanity/proxy-http.test.js]: What does server.options contain?

Comment (kunagpal) [inline test/integration/sanity/proxy-http.test.js]:
```json
{
   "target": "<url>",
   "prependPath": true
}
```

File: package.json (hunk 1 above)
Comment (kunagpal) [inline package.json]: Alright, I'll shift it in a subsequent commit.

File: test/integration/sanity/proxy-http.test.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+describe('sanity test', function () {
+    var proxy = require('http-proxy'),
+
+        server = new proxy.createProxyServer({target: 'http://echo.getpostman.com', headers: {
```
Comment (czardoz) [inline test/integration/sanity/proxy-http.test.js]: indentation.

File: test/integration/sanity/proxy-http.test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                    proxy: 'http://localhost:9090',
+                    tunnel: false
+                });
+            }
+        };
+
+        server.listen(9090);
+
+        this.run({
+            collection: {
```
Comment (czardoz) [inline test/integration/sanity/proxy-http.test.js]: Indentation

File: test/integration/sanity/proxy-http.test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        expect(testrun).be.ok();
+        expect(testrun.done.calledOnce).be.ok();
+        expect(testrun.start.calledOnce).be.ok();
+    });
+
+    it('must receive response from the proxy', function () {
+        var response = testrun.request.getCall(0).args[2];
+
+        expect(testrun.request.calledOnce).be.ok(); // one request
+        expect(response.json()).be.ok();
```
Comment (czardoz) [inline test/integration/sanity/proxy-http.test.js]: response.json() should be a variable.

File: test/integration/sanity/proxy-http.test.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        expect(response.json().headers).to.be.ok();
+        expect(response.json().headers['x-postman-proxy']).to.be('true');
+    });
+
+    after(function () {
+        server.close();
```
Comment (czardoz) [inline test/integration/sanity/proxy-http.test.js]: Correctly close this by listening to the `close` event.

File: test/integration/sanity/proxy-http.test.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
+describe('sanity test', function () {
+    var _ = require('lodash'),
+        proxy = require('http-proxy'),
+
+        server = new proxy.createProxyServer({
```
Comment (shamasis) [inline test/integration/sanity/proxy-http.test.js]: move proxy instantiation in before block

Comment (kunagpal) [inline test/integration/sanity/proxy-http.test.js]: Done

File: test/integration/sanity/proxy-http.test.js (hunk 5 above)
Comment (kunagpal) [inline test/integration/sanity/proxy-http.test.js]: Done

File: test/integration/sanity/proxy-http.test.js (hunk 6 above)
Comment (kunagpal) [inline test/integration/sanity/proxy-http.test.js]: Done

File: test/integration/sanity/proxy-http.test.js (hunk 3 above)
Comment (kunagpal) [inline test/integration/sanity/proxy-http.test.js]: Fixed

File: test/integration/sanity/proxy-http.test.js (hunk 2 above)
Comment (kunagpal) [inline test/integration/sanity/proxy-http.test.js]: Test added.

**Pull request merged**